daemon/cmd/state.go

```diff
@@ -158,6 +158,14 @@ func (d *Daemon) restoreOldEndpoints(state *endpointRestoreState, clean bool) er
 		return nil
 	}
 
+	var emf endpointMetadataFetcher
+	if option.Config.EnableHighScaleIPcache {
+		emf = &uncachedEndpointMetadataFetcher{slimcli: d.clientset.Slim()}
+	} else {
+		emf = &cachedEndpointMetadataFetcher{k8sWatcher: d.k8sWatcher}
```

Comment:
nit/followup: Should these implement some metric? It'd be nice to know in future if the uncached fetcher ends up triggering frequently.

Reply:
cc @christarazi

Reply:
:+1:, can do. `cilium_k8s_uncached_pod_metadata_fetches` or similar?